### DISCUSSION TITLE
Best Resources for Learning Reverse Engineering: A Guide for Beginners and Advanced Learners

### DISCUSSION THREAD

**Discussion by vanhdayy** [community/community#140081] [Programming Help] [answered]
### Body

Could you recommend resources or tutorials for learning reverse engineering? I’m looking for both beginner-friendly and more advanced materials, especially focused on tools, techniques, and practical exercises.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by vanhbakaa** [comment, marked as answer]
A highly recommended resource for learning reverse engineering is the book "Practical Reverse Engineering" by Bruce Dang, Alexandre Gazet, Elias Bachaalany, and Sebastien Josse.

It covers the basics and more advanced techniques, using real-world examples and tools like IDA Pro, OllyDbg, and WinDbg. It’s suitable for both beginners and intermediate learners, providing a solid foundation in reverse engineering.
